img_merger.py

```diff
@@ -0,0 +1,229 @@
+parser.add_argument('--cvat_annot_file', help='path to file for cvat annotations', default='cvat_images/originals/cvat_annot_coco.json', type=str)
+parser.add_argument('--biggie_annot_file', help='path to file for biggie annotations', default='cvat_images/originals/biggie_annot_coco.json', type=str)
+parser.add_argument('--vis_extension', help='extension for visualized smallies and annotated image output', default='jpg', type=str)
+args = parser.parse_args()
+
+f2b_filepath = Path(args.f2b_file)
+assert f2b_filepath.is_file(), 'img data file doesnt exist'
+with open(str(f2b_filepath)) as json_file:
+    f2b_data = json.load(json_file)
+
+biggie_annots_path = Path(args.biggie_annot_file)
+
+smol_annots_dir = f2b_filepath.parent / 'smol_annots'
+Path.mkdir(smol_annots_dir, exist_ok=True)
+
+cvat_annots_path = Path(args.cvat_annot_file)
+assert cvat_annots_path.is_file(), 'img data file doesnt exist'
+
+# check if 'coco' or 'cvat for images' format
+if cvat_annots_path.suffix == '.json':
+    assert biggie_annots_path.suffix=='.json', 'biggie_annot_file needs to be a json'
+
+    with open(str(cvat_annots_path)) as json_file:
+        cvat_annots = json.load(json_file)
+
+    biggie_annots = baseline_info_coco(cvat_annots)
+
+    # get category labels
+    cats_f2b = {}
+    for cat_dict in cvat_annots['categories']:
+        cats_f2b[cat_dict['id']] = cat_dict['name']
+
+    biggie_images = []
+    all_flatten_dets = []
+    all_smallie_annots_coco = []
+    f2b_merged_dicts = []
+    for i, img_name in enumerate(f2b_data):
+        # read original image and f2b settings
+        f2b_settings = f2b_data[img_name]
+
+        input_path = Path(f2b_settings["img_path"])
+        biggie = cv2.imread(str(input_path))
+
+        if biggie is None:
+            print(f'{input_path} does not exist')
+            continue
+
+        biggie_images.append({"id": i+1, "width": biggie.shape[1], "height": biggie.shape[0], "file_name": img_name, "license": 0})
+
+        f2b = F2B(
+            max_inference_width = f2b_settings['max_inference_width'],
+            max_inference_height = f2b_settings['max_inference_height'],
+            overlapx_px = f2b_settings['overlapx_px'],
+            overlapy_px = f2b_settings['overlapy_px'],
+            )
+        num_smols = f2b.register(biggie.shape[:2])
+
+        # get all annotations of all smallies for the image
+        img_annots = []
+        img_smallie_annots_coco = []
+        for smallie_path in f2b_settings["smols"]:
+            # find image_id for smallie
+            smallie_name = Path(smallie_path).name
+            image_data = next((item for item in cvat_annots['images'] if item["file_name"] == smallie_name), None)
+
+            if image_data is not None:
+                # get all annotations for smallie
+                smallie_annots_coco = [element for element in cvat_annots['annotations'] if element['image_id'] == image_data['id']]
```

Comment:
This can possibly be made more efficient. Currently we need to go through the entire annotations list once for each `smallie_path`, might not scale very well for large sets. Maybe add a step earlier to just go through the annotations list once, and index each annotation to each image_id (maybe as a dict or smth, up to you). In that case, for each `smallie_path` will just be look up in the dictionary to get its list of annotations.